crates/biome_js_analyze/src/lint/nursery/use_throw_new_error.rs

```diff
@@ -0,0 +1,180 @@
+    /// ```js
+    /// throw new TypeError('biome');
+    /// ```
+    /// ```js
+    /// throw new lib.TypeError();
+    /// ```
+    ///
+    pub UseThrowNewError {
+        version: "next",
+        name: "useThrowNewError",
+        sources: &[RuleSource::EslintUnicorn("throw-new-error")],
+        recommended: false,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+impl Rule for UseThrowNewError {
+    type Query = Ast<JsCallExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if is_in_throw_statement(node.clone()).not() {
+            return None;
+        }
+
+        let callee = &node.callee().ok()?.omit_parentheses();
+        let name = match callee {
+            AnyJsExpression::JsIdentifierExpression(ident_expr) => Some(ident_expr.text()),
+            AnyJsExpression::JsStaticMemberExpression(member_expr) => {
+                let member_expr = member_expr.member().ok()?.text();
+
+                Some(member_expr)
+            }
+            _ => None,
+        }?;
+
+        if ERROR_CONSTRUCTOR_REGEX.is_match(&name).not() {
+            return None;
+        }
+
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {
+                "Use "<Emphasis>"new"</Emphasis>" when throwing an error."
+            },
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let mut mutation = ctx.root().begin();
+
+        let new_expression = convert_call_expression_to_new_expression(node)?;
+
+        mutation.replace_node::<AnyJsExpression>(node.clone().into(), new_expression.into());
+
+        Some(JsRuleAction {
+            category: ActionCategory::QuickFix,
+            applicability: Applicability::MaybeIncorrect,
+            message: markup! { "Add "<Emphasis>"new"</Emphasis>" keyword." }.to_owned(),
+            mutation,
+        })
+    }
+}
+
+lazy_static! {
+    static ref ERROR_CONSTRUCTOR_REGEX: Regex = Regex::new(r#"^(?:[A-Z][\da-z]*)*Error$"#).unwrap();
+}
+
+fn does_member_contains_call_expression(expr: &AnyJsExpression) -> Option<bool> {
+    let mut current_node = expr.clone();
+
+    loop {
+        current_node = current_node.omit_parentheses();
+
+        if let Some(static_member_expr) = current_node.as_js_static_member_expression() {
+            current_node = static_member_expr.object().ok()?.clone();
+        } else if current_node.as_js_call_expression().is_some() {
+            return Some(true);
+        } else {
+            return None;
+        }
+    }
+}
+
+fn convert_call_expression_to_new_expression(expr: &JsCallExpression) -> Option<JsNewExpression> {
```

Comment:
use_consistent_builtin_instantiation.rs also contains this function, but (again) I am not actually sure where to put them. First idea 
`crates/biome_js_syntax/src/expr_ext.rs`, but that does not have make:: facotry functions so maybe some linter utils?

Reply:
Yes this could be placed in the linter's utils module if you think the function could be used in more rules.
Otherwise, if you think that the function is too specific, you can just make the function `pub(crate)` in a rule and import it in the other rule.